spot_hardware_interface/src/spot_hardware_interface.cpp

```diff
@@ -65,6 +82,29 @@ hardware_interface::CallbackReturn SpotHardware::on_init(const hardware_interfac
 
   njoints_ = hw_states_.size() / interfaces_per_joint_;
 
+  // check that the number of joints matches what we expect, and determine default k_q_p/k_qd_p from this
+  std::vector<float> default_k_q_p, default_k_qd_p;
+  if (njoints_ == kNjointsArm) {
+    default_k_q_p.assign(std::begin(kDefaultKqpArm), std::end(kDefaultKqpArm));
+    default_k_qd_p.assign(std::begin(kDefaultKqdpArm), std::end(kDefaultKqdpArm));
+  } else if (njoints_ == kNjointsNoArm) {
+    default_k_q_p.assign(std::begin(kDefaultKqpNoArm), std::end(kDefaultKqpNoArm));
+    default_k_qd_p.assign(std::begin(kDefaultKqdpNoArm), std::end(kDefaultKqdpNoArm));
+  } else {
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"),
+                 "Got %ld joints, expected either %d (Spot with arm) or %d (Spot without arm)!!", njoints_, kNjointsArm,
+                 kNjointsNoArm);
+    return hardware_interface::CallbackReturn::ERROR;
+  }
+
+  // Get the user-passed in k_q_p and k_q_d values.
+  const std::string k_q_p_string = info_.hardware_parameters["k_q_p"];
+  const std::string k_qd_p_string = info_.hardware_parameters["k_qd_p"];
```

Comment:
@khughes-bdai what if either parameter is not defined?

Reply:
If either isn't defined, these values are just the empty string. It ends up going through the rest of the conversion pipeline as an empty vector, which is then ignored in favor of the default gains since it doesn't have the right number of elements.